### PR TITLE
added refactor solving problem 7

### PR DESCRIPTION
**Pull request by alexmbentley**

### CODE REVIEW

File: models/games-model.js [hunk 1]
```diff
@@ -7,11 +7,26 @@ exports.readCategories = () => {
...
-        return rows[0];
+        for (num of rows) {
+          if (rows.length === 1) {
+            num.comment_count = 0;
+            return rows[0];
+          } else if (rows.length > 1) {
+            num.comment_count = rows.length;
+          }
+          return rows[0];
+        }
```
Comment (emilyb93) [inline models/games-model.js:29]: Would rather you did this in a single SQL statment. we have joined the comments table correctly, but we can use this now to take advantage of the SQL COUNT() function. Have a look at the notes here in the Group By and aggregate functions section https://notes.northcoders.com/courses/js-back-end/sql-relationships

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Refactor to a single query

File: models/games-model.js [hunk 2]
```diff
@@ -7,26 +7,19 @@ exports.readCategories = () => {
...
-          } else if (rows.length > 1) {
-            num.comment_count = rows.length;
-          }
-          return rows[0];
-        }
+    .then((data) => {
+      newData = data.rows;
+      let result = newData.find((item) => item.review_id == id);
+      if (result !== undefined) {
+        result.comment_count = parseInt(result.comment_count);
```
Comment (emilyb93) [inline models/games-model.js:21]: We can do this in SQL as well! After your COUNT() function, we can cast the data into a certain type 
You can do this by adding ```:: INT``` after your COUNT()

**Review by emilyb93 — APPROVED** [review]
Good stuff so far! Would like to see that type cast in SQL but up to you

File: models/games-model.js (hunk 2 above)
Comment (alexmbentley) [reply models/games-model.js:21]: Thanks! will add and merge

**Pull request merged**